lib/matrix_convert_to_zero.rb

```diff
@@ -3,8 +3,71 @@
 # If any number is found to be 0, the method updates all the numbers in the
 # corresponding row as well as the corresponding column to be 0.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(r * c), where r equals the number of arrays (rows) inside the array, and
+#                    c equals the number of elements (columns) inside of a nested array
+# Space complexity: O(1), or constant.  We're not creating any additional data structures that hinges on user input
 def matrix_convert_to_zero(matrix)
-  raise NotImplementedError
+  # Check for edge cases
+  return if (matrix == nil || matrix == [])
+
+  # Determine length of rows and columns
+  rows = matrix.length
+  columns = matrix[0].length
+
+  row_zero_zeroes = false
+  # Does Row 0 have zeros?
+  columns.times do |col|
+    if matrix[0][col] == 0
+      row_zero_zeroes = true
+    end
+  end
+
+  column_zero_zeroes = false
```

Comment:
I'd recommend having all of your variables at the top of your method, that way if you ever need to add or change a variable you do not need to hunt through the method to find it.